src/lib/containers/FilePreview/HtmlPreview/HtmlPreview.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useMemo } from 'react'
+import xss from 'xss'
+import { xssOptions } from '../../../utils/functions/SanitizeHtmlUtils'
+import { useGetTeamMembers } from '../../../utils/hooks/SynapseAPI/team/useTeamMembers'
+import { TRUSTED_HTML_USERS_TEAM_ID } from '../../../utils/SynapseConstants'
+import { FileHandle } from '../../../utils/synapseTypes'
+import { SynapseSpinner } from '../../LoadingScreen'
+import { SanitizedWarning } from './SanitizedWarning'
+
+/**
+ * @param options
+ * @returns HTML that is cleaned if it is not trusted. undefined if `isLoading` is true.
+ */
+function useCleanHtml(options: {
+  rawHtml: string
+  isLoading: boolean
+  isTrusted: boolean
+}): string | undefined {
+  const { rawHtml, isLoading, isTrusted } = options
+  return useMemo(() => {
+    if (isLoading) {
+      return undefined
+    }
+    if (isTrusted) {
+      return rawHtml
+    } else {
+      return xss(rawHtml, xssOptions)
+    }
+  }, [isLoading, isTrusted, rawHtml])
+}
+
+export type HtmlPreviewProps = {
+  fileHandle: FileHandle
+  rawHtml: string
+}
+
+/**
+ * Renders raw HTML. Uses file handle data to determine if the content should be sanitized.
+ * @param props
+ * @returns
+ */
+export default function HtmlPreview(props: HtmlPreviewProps) {
+  const { fileHandle, rawHtml } = props
+  const { data: trustedHtmlUsersData, isLoading } = useGetTeamMembers(
```

Comment:
Will this get the first 50 members only?

Reply:
I think it's hard-capped at 20. Not an issue today since the team is not that large, but I can update this to fetch all members

Reply:
The team has 22 members so if it is hard-capped at 20, it will cause an issue today

Reply:
But the cap is indeed 50, so no issue. Still worth changing.